Source/Diagnostics/BTDiagnostics.cpp

```diff
@@ -465,15 +478,23 @@ BTDiagnostics::k_index_zlab (int i_buffer, int lev)
     amrex::Real prob_domain_zmin_lab = m_prob_domain_lab[i_buffer].lo( m_moving_window_dir );
     amrex::IntVect ref_ratio = amrex::IntVect(1);
     if (lev > 0 ) ref_ratio = WarpX::RefRatio(lev-1);
-    int k_lab = static_cast<unsigned>( (
+    int k_lab = static_cast<int>(floor (
```

Comment:
Changed this to floor so that the same k-index is not filled twice.
For example, when the double for this valud was ~0.25 or ~-0.25, it would round-off to 0, over-writing the previously filled 0th index. 
Please let me know if you have other suggestions